### ISSUE TITLE
can not verify and publish contract code, Constructor arguments do not match, please try again.

### ISSUE THREAD

**Issue by passionofvc** [blockscout/blockscout#3331]
blockscout cat not verify and publish contract code, 
[Constructor arguments do not match, please try again].
When I  set [Try to fetch contructor arguments automatically] to [Yes], the result is also
[Constructor arguments do not match, please try again].
 
### Environment
* Elixir & Erlang/OTP versions (`elixir -version`):
elixir -version
* Operating System:
Ubuntu 20.04.1 LTS
* Blockscout Version/branch: 
BlockScout v3.3.3-beta

### Steps to reproduce

1.  run file-flat_compile-sh
  https://gist.github.com/passionofvc/79dbde68da461a88d14d080c36fcc350#file-flat_compile-sh

2. run deploy contract  
 https://gist.github.com/passionofvc/79dbde68da461a88d14d080c36fcc350#file-deploy_contract-js

3. start blockscout from docker 
  git clone https://github.com/poanetwork/blockscout.git
  cd docker && make start

4. go to blockscout verify and publish code use below input
```
| Contract Name                     | Diamond                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                          |   |   |   |
|-----------------------------------|------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|---|---|---|
| Compiler                          | 0.7.1+commit.f4a555be                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            |   |   |   |
| EVM Version                       | istanbul                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         |   |   |   |
| Optimization                      | Yes                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              |   |   |   |
| Optimization runs                 | 200                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                              |   |   |   |
| ABI-encoded Constructor Arguments | 000000000000000000000000000000000000000000000000000000000000004000000000000000000000000073b4fb6848ce26872a82f86147ffeeeb6cea4ea20000000000000000000000000000000000000000000000000000000000000002000000000000000000000000000000000000000000000000000000000000004000000000000000000000000000000000000000000000000000000000000000e00000000000000000000000000280666a6911dee91a9c178081cea4b8b39a992d0000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000006000000000000000000000000000000000000000000000000000000000000000011f931c1c00000000000000000000000000000000000000000000000000000000000000000000000000000000a57fbc493acc9affb0174791e80d2d21563408db0000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000000006000000000000000000000000000000000000000000000000000000000000000057a0ed62700000000000000000000000000000000000000000000000000000000adfca15e0000000000000000000000000000000000000000000000000000000052ef6b2c00000000000000000000000000000000000000000000000000000000cdffacc60000000000000000000000000000000000000000000000000000000001ffc9a700000000000000000000000000000000000000000000000000000000 |   |   |   |
```

### Expected behaviour
Should verify and publish Diamond's contract code,

### Actual behaviour
always show [Constructor arguments do not match, please try again].
can not add contract code, when I delete all construtor's paramater, it is verified ok.

**Comment by heri16** [comment]
I'm having the same problem on Mumbai explorer matic vigil when Verifying a proxy contract. All other contracts verified fine.

**Comment by iaryama** [comment]
@heri16 I am facing the same issue in Mumbai explorer Matic. Getting the same error.

**Comment by royki** [comment]
Hi @passionofvc @vbaranov 
Is it any resolution - 

We have deployed this smart contract in Volta network - 
- https://github.com/energywebfoundation/origin/blob/master/packages/traceability/issuer/contracts/RegistryExtended.sol
- https://volta-explorer.energyweb.org/address/0xE985cac6C713C54f5b0CAcd18320029114C797a7/contracts
While trying to verify and publish, getting same error - `Constructor arguments do not match, please try again.`  

If you need the flatten file, I can provide. Please let me know how to overcome this.

**Comment by vbaranov** [comment]
@passionofvc @royki there is a fix for that in 4.0.0 https://github.com/blockscout/blockscout/pull/4764
